e2e/network_test.go

```diff
@@ -0,0 +1,284 @@
+package e2e
+
+import (
+	"fmt"
+	"strings"
+	"testing"
+	"time"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/pkg/api"
+	"k8s.io/client-go/pkg/api/v1"
+	"k8s.io/client-go/pkg/apis/extensions/v1beta1"
+)
+
+func TestNetwork(t *testing.T) {
+	//
+	//
+	//
+	// 1. create nginx service
+	di, _, err := api.Codecs.UniversalDecoder().Decode(nginxDepNT, nil, &v1beta1.Deployment{})
+	if err != nil {
+		t.Fatalf("unable to decode deployment manifest: %v", err)
+	}
+
+	d, ok := di.(*v1beta1.Deployment)
+	if !ok {
+		t.Fatalf("expected manifest to decode into *api.deployment, got %T", di)
+	}
+	_, err = client.ExtensionsV1beta1().Deployments(namespace).Create(d)
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer client.ExtensionsV1beta1().Deployments(namespace).Delete("nginx-deployment-nt", &metav1.DeleteOptions{})
+
+	if err := retry(10, time.Second*10, getNginxPod); err != nil {
+		t.Fatalf("timed out waiting for nginx pod: %v", err)
+	}
+
+	si, _, err := api.Codecs.UniversalDecoder().Decode(nginxSVC, nil, &v1.Service{})
+	if err != nil {
+		t.Fatalf("unable to decode service manifest: %v", err)
+	}
+	s, ok := si.(*v1.Service)
+	if !ok {
+		t.Fatalf("expected manifest to decode into *api.service, got %T", si)
+	}
+	_, err = client.CoreV1().Services(namespace).Create(s)
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer client.CoreV1().Services(namespace).Delete("nginx-service", &metav1.DeleteOptions{})
+
+	//
+	//
+	//
+	// 2. create a wget pod that hits the nginx service
+	_, err = client.CoreV1().Pods(namespace).Create(wgetPodNT)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	if err := retry(10, time.Second*10, getPod); err != nil {
+		t.Fatalf(fmt.Sprintf("timed out waiting for wget pod to succeed: %v", err))
+	}
+	client.CoreV1().Pods(namespace).Delete(wgetPodNT.ObjectMeta.Name, &metav1.DeleteOptions{})
+
+	fmt.Println("Starting default deny test")
```

Comment:
These two Println statements seems like separate tests within a single test. Might be worth making use of go's subtests here. 

Regardless, its better form to log via the test logger then just printing to stdout directly.